js/src/as-report.js

```diff
@@ -4,45 +4,60 @@
  */
 import { __ } from '@wordpress/i18n';
 import apiFetch from '@wordpress/api-fetch';
-import { Component } from '@wordpress/element';
-import { Fragment } from '@wordpress/element';
-import { Card, EmptyContent, ReportFilters, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { Component, Fragment } from '@wordpress/element';
 import { map } from 'lodash';
 
+/**
+ * WooCommerce dependencies
+ */
+import { Card, EmptyContent, ReportFilters, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { onQueryChange, stringifyQuery } from '@woocommerce/navigation';
+
 /**
  * Internal dependencies
  */
+import { statusFilters } from './config';
 const showDatePicker = false;
 
 //@todo: import './style.scss'; if necessary
 
 class ActionsReport extends Component {
 
-	constructor() {
-		super();
+	constructor( props ) {
+		super( props );
 		this.state = {
 			actions: null,
 			loading: true,
 		};
+		this.getHeadersContent = this.getHeadersContent.bind( this );
+		this.getRowsContent = this.getRowsContent.bind( this );
 	}
 
 	componentDidMount() {
 		// This should be handled for us automagically, but the nonce wasn't working for me
 		apiFetch.use( apiFetch.createNonceMiddleware( asaSettings.nonce ) );
 		apiFetch.use( apiFetch.createRootURLMiddleware( asaSettings.api_root ) );
-// @todo: add query parameters
-		this.fetchActionData();
+
+		this.fetchActionData( {
+			orderby: 'scheduled',
+			order: 'asc',
+			status: 'pending',
+		} );
 	}
 
 	componentDidUpdate( prevProps ) {
 		const prevQuery = prevProps.query;
 		const query = this.props.query;
+
+		if (  query != prevQuery ) {
+			this.fetchActionData( query );
```

Comment:
Requests without `order` or `orderBy` are 500'ing for me. In wc-admin we add defaults here, but you could handle on the backend as well.

Reply:
I get the full result set when using a REST client without a query string or just `?status=pending`. Do you have debug output enabled?

Reply:
I'm getting nothing on `debug.log` but I suspect I'm experiencing a pagination issue due to the number of jobs I have:

![Screen Shot 2019-05-09 at 9 06 43 AM](https://user-images.githubusercontent.com/1922453/57408500-e60fac80-7239-11e9-8f3f-47ef12e97a82.png)

Reply:
Added #7 for this.